test/models/ride_test.dart

```diff
@@ -0,0 +1,468 @@
+      expect(ride.createdAt, DateTime.parse(json['created_at']));
+      expect(ride.start, json['start']);
+      expect(ride.startPosition, Position.fromDynamicValues(json['start_lat'], json['start_lng']));
+      expect(ride.startTime, DateTime.parse(json['start_time']));
+      expect(ride.end, json['end']);
+      expect(ride.endPosition, Position.fromDynamicValues(json['end_lat'], json['end_lng']));
+      expect(ride.endTime, DateTime.parse(json['end_time']));
+      expect(ride.seats, json['seats']);
+      expect(ride.price, parseHelper.parseDouble(json['price']));
+      expect(ride.status, RideStatus.values[json['status']]);
+      expect(ride.riderId, json['rider_id']);
+      expect(ride.driveId, json['drive_id']);
+      expect(ride.drive!.id, json['drive_id']);
+      expect(ride.rider!.id, json['rider_id']);
+      expect(ride.hideInListView, json['hide_in_list_view']);
+    });
+  });
+  group('Ride.fromJsonList', () {
+    test('parses a list of drives from json', () {
+      final Map<String, dynamic> json = {
+        'id': 43,
+        'created_at': '2021-01-01T00:00:00.000Z',
+        'start': 'London',
+        'start_lat': 2,
+        'start_lng': 3,
+        'start_time': '2022-01-01T00:00:00.000Z',
+        'end': 'Berlin',
+        'end_lat': 4.0,
+        'end_lng': 3.0,
+        'end_time': '2023-01-01T00:00:00.000Z',
+        'seats': 2,
+        'price': 3.5,
+        'status': 3,
+        'drive_id': 7,
+        'rider_id': 4,
+        'hide_in_list_view': false,
+      };
+      final List<Map<String, dynamic>> jsonList = [json, json, json];
+      final List<Ride> rides = Ride.fromJsonList(jsonList);
+      expect(rides.length, 3);
+      expect(rides.first.id, json['id']);
+      expect(rides[1].start, json['start']);
+      expect(rides.last.seats, 2);
+    });
+    test('can handle an empty list', () {
+      final List<Ride> drives = Ride.fromJsonList([]);
+      expect(drives, []);
+    });
+  });
+  group('Ride.toJson', () {
+    test('returns a json representation of the ride', () async {
+      final Ride ride = RideFactory().generateFake();
+      final Map<String, dynamic> json = ride.toJson();
+      expect(json['start'], ride.start);
+      expect(json['start_lat'], ride.startPosition.lat);
+      expect(json['start_lng'], ride.startPosition.lng);
+      expect(json['start_time'], ride.startTime.toString());
+      expect(json['end'], ride.end);
+      expect(json['end_lat'], ride.endPosition.lat);
+      expect(json['end_lng'], ride.endPosition.lng);
+      expect(json['end_time'], ride.endTime.toString());
+      expect(json['status'], ride.status.index);
+      expect(json['seats'], ride.seats);
+      expect(json['price'], ride.price);
+      expect(json['drive_id'], ride.driveId);
+      expect(json['rider_id'], ride.riderId);
+      expect(json['hide_in_list_view'], ride.hideInListView);
+      expect(json.keys.length, 14);
+    });
+  });
+  group('Ride.toJsonforApi', () {
```

Comment:
Do we test this now? I think tests for testing infrastructure are a bit much, we're already getting the coverage by calling this 100 times in tests anyway.

Reply:
I can take it out if you want. But I don't think it hurts